knox/views.py

```diff
@@ -79,3 +79,33 @@ def post(self, request, format=None):
         user_logged_out.send(sender=request.user.__class__,
                              request=request, user=request.user)
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+
+class TokenRefreshView(APIView):
+
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, format=None):
+        return Response(
+            {'expiry': request.auth.expiry},
+            status=status.HTTP_200_OK
+        )
+
+
+class TokenRefreshAllView(APIView):
+
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, format=None):
+        request.user.auth_token_set.all().update(
+            expiry=timezone.now() + knox_settings.TOKEN_TTL
+        )
+        """either we return expiry for 1 key which
+        is the same for all keys, or we return nothing
+        and set response code 204 indicating success,
+        but no data returned"""
+        return Response(
```

Comment:
I am not sure what makes most sense here. We update potentially multiple instances, do we just return `expiry` and imply its the new time for all tokens.. or do we just return 204?

Reply:
It's probably more consistent to return the `expiry`.
I agree this boils to implying something but at the same time we just introduced the fact that the login view would return the expiry and the refresh/token view also returns the expiry.
I personally would be surprised not having the expiry returned.

Reply:
I agree with @johnraz